src/lib/structures/Middleware.ts

```diff
@@ -58,12 +58,18 @@ export class Middleware<StoredValue = unknown> {
    */
   public readonly conditions: Middleware.Conditions;
 
-  public constructor(options: Middleware.Options) {
+  protected context: ContextData;
+
+  public constructor(context: ContextData, options: Partial<Middleware.Options> = {}) {
+    this.context = context;
+
     const { name, position, conditions } = options;
 
+    if (!name) throw new JoshError({ identifier: Middleware.Identifiers.NameNotFound, message: 'No name was provided for this middleware.' });
```

Comment:
Needs to be changed due to #57

Reply:
There is no `error()` method on `Middleware`

Reply:
Ah yep I see, coolio